xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1532,26 +1528,21 @@ bool CGUIMediaWindow::OnPlayAndQueueMedia(const CFileItemPtr& item, const std::s
     // Remove ZIP, RAR files and folders
     CFileItemList playlist;
     playlist.Copy(*m_vecItems, true);
+
     playlist.erase(std::remove_if(playlist.begin(), playlist.end(),
                                   [](const std::shared_ptr<CFileItem>& i)
                                   { return i->IsZIP() || i->IsRAR() || i->m_bIsFolder; }),
                    playlist.end());
 
-    // Remove duplicates (eg. ISO/VIDEO_TS)
-    playlist.erase(
-        std::unique(playlist.begin(), playlist.end(),
-                    [](const std::shared_ptr<CFileItem>& i, const std::shared_ptr<CFileItem>& j) {
-                      return i->GetVideoInfoTag()->m_basePath == j->GetVideoInfoTag()->m_basePath;
-                    }),
-        playlist.end());
-
     // Chosen item
     int mediaToPlay =
         std::distance(playlist.begin(), std::find_if(playlist.begin(), playlist.end(),
-                                                     [&item](const std::shared_ptr<CFileItem>& i) {
-                                                       return i->GetVideoInfoTag()->m_basePath ==
-                                                              item->GetVideoInfoTag()->m_basePath;
-                                                     }));
+                                                     [&item](const std::shared_ptr<CFileItem>& i)
+                                                     { return i->GetPath() == item->GetPath(); }));
```

Comment:
Are you sure the change to not compare `m_basepath` but `Path()`is still needed?

Reply:
Yes. That was part of the problem - if you select an item from Video -> Files that isn't in the library then the VideoInfoTag and hence m_basePath were empty.

Reply:
Okay.